File: sdk/python/examples/sovereign.py
"""Sovereign AI — RGPD-compliant EU-only routing.

Forces all requests through EU providers (Scaleway, OVHCloud)
to guarantee data sovereignty.  PII is automatically detected
server-side and routed to EU providers even without this mode.

Run:
    export AI_INFRA_API_KEY="sk-your-key-here"
    python examples/sovereign.py
"""

from ai_infra import Client

# Create client with EU-only routing mode
client = Client(mode="eu_only")

# This request will ONLY use EU providers
response = client.chat.completions.create(
    model="auto",
    messages=[
        {"role": "user", "content": "Analyse ce contrat de travail pour Jean Dupont."},
    ],
    pii_check=True,  # Warn client-side if PII detected
)

print(response.choices[0].message.content)

# Verify sovereignty
print(f"\nProvider: {response.savings.provider_used}")
print(f"RGPD compliant: {response.savings.rgpd_compliant}")
print(f"EU routing: {response.savings.eu_routing}")
print(f"Forced EU routing (PII): {response.savings.forced_eu_routing}")
print(f"PII types detected: {response.savings.pii_types_detected}")
print(f"Cache hit: {response.savings.cache_hit}")
print(f"Model: {response.savings.model_used}")

# List EU-safe models
print("\nEU-safe models available:")
for model in client.list_models().data:
    if model.eu_safe:
        print(f"  - {model.id}")

File: sdk/typescript/examples/sovereign.ts
/**
 * Sovereign AI — RGPD-compliant routing with PII detection.
 *
 * When PII is detected, the platform automatically routes to EU-only
 * providers (Scaleway, OVHCloud) with EU-safe models (Mistral, Llama, Gemma).
 *
 * Prerequisites:
 *   export AI_INFRA_API_KEY="sk-your-key-here"
 *   npx ts-node examples/sovereign.ts
 */

import { Client, RoutingMode, AIInfraError } from "../src";

async function main() {
  const client = new Client({
    mode: RoutingMode.EuOnly, // Force EU routing for all requests
    telemetry: true,
  });

  try {
    // Request with PII — server-side Presidio will detect it
    // and enforce EU-only routing automatically
    const response = await client.chat.completions.create({
      messages: [
        { role: "user", content: "Résume le dossier médical de Jean Dupont, né le 15/03/1985, IBAN FR76 1234 5678 9012" },
      ],
      pii_check: true, // Enable client-side PII warning
    });

    console.log(response.choices[0].message.content);
    console.log();
    console.log("--- Sovereignty Report ---");
    console.log(`EU routing:        ${response.savings.eu_routing}`);
    console.log(`Forced EU routing: ${response.savings.forced_eu_routing}`);
    console.log(`RGPD compliant:    ${response.savings.rgpd_compliant}`);
    console.log(`PII types found:   ${response.savings.pii_types_detected.join(", ") || "none"}`);
    console.log(`Provider:          ${response.savings.provider_used}`);
    console.log(`Model:             ${response.savings.model_used}`);
  } catch (error) {
    if (error instanceof AIInfraError) {
      console.error(`API Error: ${error.message} (status: ${error.statusCode})`);
    } else {
      throw error;
    }
  }

  // Check telemetry stats
  const stats = client.getStats();
  console.log();
  console.log("--- Telemetry ---");
  console.log(`Total requests: ${stats.totalRequests}`);
  console.log(`Total cost:     $${stats.totalCostUsd.toFixed(4)}`);
  console.log(`Total saved:    $${stats.totalSavingsUsd.toFixed(4)}`);
}

main().catch(console.error);
